sdk/process/backend.go

```diff
@@ -0,0 +1,146 @@
+package processsdk
+
+import (
+	"errors"
+	"fmt"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/mesg-foundation/engine/codec"
+	"github.com/mesg-foundation/engine/cosmos"
+	"github.com/mesg-foundation/engine/hash"
+	"github.com/mesg-foundation/engine/ownership"
+	"github.com/mesg-foundation/engine/process"
+	instancesdk "github.com/mesg-foundation/engine/sdk/instance"
+	ownershipsdk "github.com/mesg-foundation/engine/sdk/ownership"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+const backendName = "process"
+
+// Backend is the service backend.
+type Backend struct {
+	storeKey  *cosmostypes.KVStoreKey
+	ownership *ownershipsdk.Backend
+	instance  *instancesdk.Backend
+}
+
+// NewBackend returns the backend of the service sdk.
+func NewBackend(appFactory *cosmos.AppFactory, ownership *ownershipsdk.Backend, instance *instancesdk.Backend) *Backend {
+	backend := &Backend{
+		storeKey:  cosmostypes.NewKVStoreKey(backendName),
+		ownership: ownership,
+		instance:  instance,
+	}
+	appBackendBasic := cosmos.NewAppModuleBasic(backendName)
+	appBackend := cosmos.NewAppModule(appBackendBasic, backend.handler, backend.querier)
+	appFactory.RegisterModule(appBackend)
+	appFactory.RegisterStoreKey(backend.storeKey)
+	return backend
+}
+
+func (s *Backend) handler(req cosmostypes.Request, msg cosmostypes.Msg) (hash.Hash, error) {
+	switch msg := msg.(type) {
+	case msgCreateProcess:
+		p, err := s.Create(req, &msg)
+		if err != nil {
+			return nil, err
+		}
+		return p.Hash, nil
+	case msgDeleteProcess:
+		if err := s.ownership.Delete(req, msg.Owner, msg.Request.Hash); err != nil {
```

Comment:
please create a dedicated function `Delete` even if it's small. Let's not put business logic directly in the handler.